sandy/core/cov.py

```diff
@@ -341,6 +343,100 @@ def corr(self):
                             columns=self.data.columns,
                             )
 
+    def _reduce_size(self):
+        """
+        Reduces the size of the matrix, erasing the null values.
+
+        Returns
+        -------
+        nonzero_idxs : `numpy.ndarray`
+            The indices of the diagonal that are not null.
+        cov_reduced : `pandas.DataFrame`
+            The reduced matrix.
+
+        Examples
+        --------
+        >>> S = sandy.CategoryCov(np.diag(np.array([1, 2, 3])))
+        >>> non_zero_index, reduce_matrix = S._reduce_size()
+        >>> non_zero_index
+        array([0, 1, 2], dtype=int64)
+        >>> reduce_matrix
+                      0	          1	          2
+        0	1.00000e+00	0.00000e+00	0.00000e+00
+        1	0.00000e+00	2.00000e+00	0.00000e+00
+        2	0.00000e+00	0.00000e+00	3.00000e+00
+
+        >>> S = sandy.CategoryCov(np.diag(np.array([0, 2, 3])))
+        >>> non_zero_index, reduce_matrix = S._reduce_size()
+        >>> non_zero_index
+        array([1, 2], dtype=int64)
+        >>> reduce_matrix
+                      1	          2
+        1	2.00000e+00	0.00000e+00
+        2	0.00000e+00	3.00000e+00
+        """
+        nonzero_idxs = np.flatnonzero(np.diag(self.data))
+        cov_reduced = self.data.loc[nonzero_idxs, nonzero_idxs]
+        return nonzero_idxs, cov_reduced
+
+    def _restore_size(cls, nonzero_idxs, cov_reduced, dim):
+        """
+        Restore the size of the matrix
+
+        Parameters
+        ----------
+        nonzero_idxs : numpy.ndarray
+            The indices of the diagonal that are not null.
+        cov_reduced : sandy.core.cov._Cov
+            The reduced matrix.
+        dim : int
+            Dimension of the original matrix.
+
+        Returns
+        -------
+        cov : sandy.core.cov._Cov

```

Comment:
no, this is a `np.ndarray`, I guess

Reply:
but it can be something else...maybe a `CategoryCov `. In that case it should be a classmethod